Signafx exporter mutates original log attributes causing collector crashes

**Describe the bug**
Signafx exporter mutates original log attributes which is causing collector to crash when there are more than one exporter in the pipeline. Code parts where the mutation happens: 
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/abd11c161b5f85be047258453f3f897acb3741f4/exporter/signalfxexporter/internal/translation/logdata_to_signalfxv2.go#L61
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/abd11c161b5f85be047258453f3f897acb3741f4/exporter/signalfxexporter/internal/translation/logdata_to_signalfxv2.go#L79

**Steps to reproduce**
Setup a logs pipeline with signalfx exporter and any other exporter, for example:
```
service:
  pipelines:
    logs/signalfx-events:
      receivers: [signalfx]
      processors: [memory_limiter, batch]
      exporters: [signalfx, logging]
```

**What did you expect to see?**
No collector failures, and Signalfx events should be delivered to both exporters destinations. Collector should not mutate original logs, but instead do it on a copy or mutate translated signalfx event entity.

**What did you see instead?**
The otel-collector crashing when it receives an event in the pipeline with more than one exporter:
```
panic: runtime error: index out of range [9] with length 9

goroutine 319 [running]:
go.opentelemetry.io/collector/consumer/pdata.AttributeMap.Delete(0xc158a3b340, 0x461d646, 0x22, 0xc15809ce10)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.29.0/consumer/pdata/common.go:420 +0x173
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/translation.convertLogRecord(0xc158a3b300, 0xc158a5cd40, 0xc15358e4b0, 0xc1669b47e0, 0x0)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter@v0.29.0/translation/logdata_to_signalfxv2.go:61 +0x4b3
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/translation.LogSliceToSignalFxV2(0xc15358e4b0, 0xc158a5cda0, 0xc158a5cd40, 0x16, 0x2030a100000000, 0x400, 0x400)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter@v0.29.0/translation/logdata_to_signalfxv2.go:37 +0xd0
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter.(*sfxEventClient).pushLogsData(0xc15358e550, 0x4d89eb8, 0xc28acb49c0, 0xc2a01c93f8, 0x0, 0x0, 0x0)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter@v0.29.0/eventclient.go:57 +0x1cc
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter.(*signalfxExporter).pushLogs(...)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter@v0.29.0/exporter.go:191
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsRequest).export(0xc1669cb880, 0x4d89eb8, 0xc28acb49c0, 0xc1669d3480, 0xc1669e1600)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.29.0/exporter/exporterhelper/logs.go:54 +0x47
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc0009c7a58, 0x4d9e558, 0xc1669cb880, 0x0, 0x0)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.29.0/exporter/exporterhelper/common.go:229 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc153584ea0, 0x4d9e558, 0xc1669cb880, 0x4585eac, 0x5)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.29.0/exporter/exporterhelper/queued_retry.go:274 +0x28a
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsExporterWithObservability).send(0xc15344baa0, 0x4d9e558, 0xc1669cb880, 0x4d9e558, 0xc1669cb880)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.29.0/exporter/exporterhelper/logs.go:116 +0xb8
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1(0x4015700, 0xc1669cb880)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/go.opentelemetry.io/collector@v0.29.0/exporter/exporterhelper/queued_retry.go:154 +0x75
github.com/jaegertracing/jaeger/pkg/queue.ConsumerFunc.Consume(0xc000c2e120, 0x4015700, 0xc1669cb880)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/jaegertracing/jaeger@v1.23.0/pkg/queue/bounded_queue.go:104 +0x3a
github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumersWithFactory.func1(0xc0009ec020, 0xc15358e5f0)
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/jaegertracing/jaeger@v1.23.0/pkg/queue/bounded_queue.go:83 +0xe9
created by github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumersWithFactory
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/jaegertracing/jaeger@v1.23.0/pkg/queue/bounded_queue.go:73 +0xc5
```

**What version did you use?**
Version: `v0.29.0`, but the bug can be reproduced on the latest build

--- a/exporter/signalfxexporter/internal/translation/logdata_to_signalfxv2.go
+++ b/exporter/signalfxexporter/internal/translation/logdata_to_signalfxv2.go
@@ -58,7 +58,6 @@ func convertLogRecord(lr pdata.LogRecord, resourceAttrs pdata.AttributeMap, logg
 	if categoryVal.Type() == pdata.AttributeValueTypeInt {
 		asCat := sfxpb.EventCategory(categoryVal.IntVal())
 		event.Category = &asCat
-		attrs.Delete(splunk.SFxEventCategoryKey)
 	}
 
 	if mapVal, ok := attrs.Get(splunk.SFxEventPropertiesKey); ok && mapVal.Type() == pdata.AttributeValueTypeMap {
@@ -76,7 +75,6 @@ func convertLogRecord(lr pdata.LogRecord, resourceAttrs pdata.AttributeMap, logg
 			return true
 		})
 	}
-	attrs.Delete(splunk.SFxEventPropertiesKey)
 
 	// keep a record of Resource attributes to add as dimensions
 	// so as not to modify LogRecord attributes
@@ -90,6 +88,11 @@ func convertLogRecord(lr pdata.LogRecord, resourceAttrs pdata.AttributeMap, logg
 	})
 
 	addDimension := func(k string, v pdata.AttributeValue) bool {
+		// Skip internal attributes
+		if k == splunk.SFxEventCategoryKey || k == splunk.SFxEventPropertiesKey {
+			return true
+		}
+
 		if v.Type() != pdata.AttributeValueTypeString {
 			logger.Debug("Failed to convert log record or resource attribute value to SignalFx property value, key is not a string", zap.String("key", k))
 			return true

